mmv1/third_party/terraform/acctest/resource_inventory_reader.go

```diff
@@ -0,0 +1,249 @@
+	// Make sure cache is populated
+	if !cachePopulated {
+		if err := PopulateMetadataCache(); err != nil {
+			return "failed_to_populate_metadata_cache"
+		}
+	}
+
+	// Check cache
+	cacheMutex.RLock()
+	servicePackage, found := servicePackageCache[resourceType]
+	cacheMutex.RUnlock()
+
+	if !found {
+		return "unknown"
+	}
+
+	return servicePackage
+}
+
+// getServicesDir returns the path to the services directory
+// If projectRoot is provided, it will use that as the base
+// Otherwise, it will attempt to find the project root, and if that fails,
+// it will use environment variables or fall back to the current directory
+func getServicesDir(projectRoot string) (string, error) {
+	// If project root is provided, use it
+	if projectRoot != "" {
+		servicesDir := filepath.Join(projectRoot, "google-beta", "services")
+		if _, err := os.Stat(servicesDir); err == nil {
+			return servicesDir, nil
+		}
+		return "", fmt.Errorf("services directory not found at %s", servicesDir)
+	}
+
+	// Try to find project root
+	root, err := findProjectRoot()
+	if err == nil {
+		servicesDir := filepath.Join(root, "google-beta", "services")
+		if _, err := os.Stat(servicesDir); err == nil {
+			return servicesDir, nil
+		}
+	}
+
+	// Try environment variable
+	if envRoot := os.Getenv("TERRAFORM_PROVIDER_ROOT"); envRoot != "" {
+		servicesDir := filepath.Join(envRoot, "google-beta", "services")
+		if _, err := os.Stat(servicesDir); err == nil {
+			return servicesDir, nil
+		}
+	}
+
+	// Last resort: try relative to current directory
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to determine current directory: %v", err)
+	}
+
+	// Try a few common relative paths
+	potentialPaths := []string{
+		filepath.Join(currentDir, "google-beta", "services"),
+		filepath.Join(currentDir, "..", "google-beta", "services"),
+		filepath.Join(currentDir, "..", "..", "google-beta", "services"),
+	}
+
+	for _, path := range potentialPaths {
+		if _, err := os.Stat(path); err == nil {
+			return path, nil
+		}
+	}
+
+	return "", fmt.Errorf("unable to locate services directory, please provide explicit project root path")
+}
+
+// findProjectRoot walks up from the current directory to find the project root
+// by looking for the go.mod file
+func findProjectRoot() (string, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	for {
+		// Check if go.mod exists in the current directory
+		if _, err := os.Stat(filepath.Join(dir, "go.mod")); err == nil {
+			return dir, nil
+		}
+
+		// Move up to the parent directory
+		parentDir := filepath.Dir(dir)
+		if parentDir == dir {
+			// Reached the filesystem root without finding go.mod
+			return "", fmt.Errorf("could not find go.mod file in any parent directory")
+		}
+		dir = parentDir
+	}
+}
+
+// ClearCache clears the internal caches for API service names and service packages
+// Useful for testing or if the underlying files have changed
+func ClearCache() {
```

Comment:
I don't see this function is called anywhere.

Reply:
removed